src/pybind/mgr/orchestrator.py

```diff
@@ -631,17 +632,18 @@ def raise_if_exception(c):
     """
     def copy_to_this_subinterpreter(r_obj):
         # This is something like `return pickle.loads(pickle.dumps(r_obj))`
-        # Without importing anything.
         r_cls = r_obj.__class__
         if r_cls.__module__ in ('__builtin__', 'builtins'):
             return r_obj
-        my_cls = getattr(sys.modules[r_cls.__module__], r_cls.__name__)
+        my_mod = importlib.import_module(r_cls.__module__)
```

Comment:
This depends on us no longer being bound by Python's reduced execution mode in some circumstances. Otherwise this is going to raise.

Actually this inner function is only required because of this reduced execution mode. Let's use pickle instead?

Reply:
what is "reduced execution mode"? is it an alias of "restricted execution mode"? if yes, will we run into https://tracker.ceph.com/issues/38407 again if we use pickle instead? probably, another option is to use something like
```py
for part in r_cls.__module__.split('.'):
    if my_mod:
        my_mod = getattr(my_mod, part)
    else:
        my_mod = sys.modules[part]
```
?

Reply:
yeah, right. restricted execution mode. I think they removed this mode in Python 3. At least I can't find any reference to https://docs.python.org/2/library/restricted.html anymore.  Either we go with pickle, or maybe even deeopcopy will create a copy from this sub-interpreter. In any case, this is then no longer Python2 compatible.

Reply:
@sebastian-philipp fixed and repushed. as this code is not sitting in critical path. i am using `copy.deepcopy()` no matter if the type is builtin or not. please note, there is chance that the underlying code throws nested types as exception, such as a `list` of user defined types. so `deepcopy()` is much simpler when handling this case.

could you take another look?